_posts/2021-02-19-26-rln-relay.md

```diff
@@ -0,0 +1,168 @@
+   1. If such message exists and its `shareX` and `shareY` components are different from the incoming message, then slashing takes place (if the `shareX` and `shareY` fields of the previously relayed message is identical to the incoming message, then the message is a duplicate and shall be dropped).
+   2. If none found, then the message gets relayed.
+
+An overview of the slashing procedure is provided in Figure 2.
+
+<p align="center">
+    <img src="../assets/img/rln-relay/rln-message-verification.png" />
+    <br />
+    Figure 2: Publishing, Routing and Slashing workflow.
+</p>
+
+
+# Feasibility and Open Issues
+
+## Storage overhead per peer
+Currently, peers are supposed to maintain the entire tree locally and it imposes storage overhead which is linear in the size of the group (see this [issue](https://github.com/vacp2p/research/issues/57) for more details). One way to cope with this is to use the light-node and full-node paradigm in which only a subset of peers who are more resourceful retain the tree whereas the light nodes obtain the necessary information by interacting with the full nodes <!--what about the anonymity issue-->. Another way to approach this problem is through a more storage efficient method (as described in this [research issue](https://github.com/vacp2p/research/pull/54)) where peers store a partial view of the tree (instead of the entire tree) which lowers the storage complexity to O(log(N)) where N is the size of the group. There are still unknown unknowns to this solution, as such, it must be studied further to become fully functional.
+
+## Cost-effective way of member insertion and deletion
+Currently, the cost associated with rln-relay membership is around 30 USD (a detailed calculation of this cost is given in this [issue](https://github.com/vacp2p/research/issues/56)). We aim at finding a more cost-effective approach. Please feel free to share with us your solution ideas in this regard in this [issue](https://github.com/vacp2p/research/issues/56). 
+
+## Exceeding the messaging rate via multiple registrations
+While the economic-incentive solution has an economic incentive to discourage spamming, we should note that there is still **Expensive attack(s)** that a spammer can launch to break the messaging rate limit. That is, the attacker can pay for multiple legit registrations e.g., k, hence being able to publish k messages per epoch. We believe that the higher the membership fee is, the less probable would be such an attack, hence a stronger level of spam-protection can be achieved. Following this argument, the high fee associated with the membership (which we listed above as an open problem) can indeed be contributing to a better protection level.  
+
+# Acknowledgement
+Thanks to Onur Kılıç for his explanation and pointers and for helping with development and runtime issues. Also thanks to Barry Whitehat for his time and insightful comments. 
+
+# References
```

Comment:
Might want to number these and refer them, I think there are some missing too re pow/whisper/peer scoring etc

Reply:
added the missing references